mne/io/tests/test_raw.py

```diff
@@ -1,14 +1,98 @@
 # Generic tests that all raw classes should run
 from os import path as op
-from numpy.testing import assert_allclose
+import math
+import numpy as np
+from numpy.testing import assert_allclose, assert_array_almost_equal
 
+from nose.tools import assert_equal, assert_true
+
+from mne import concatenate_raws
 from mne.datasets import testing
 from mne.io import Raw
+from mne.utils import _TempDir
+
+
+def _test_raw_object(reader, test_preloading, test_blocks=False, **kwargs):
+    """Test reading, writing and slicing of raw classes.
+
+    Parameters
+    ----------
+    reader : function
+        Function to test.
+    test_preloading : bool
+        Whether not preloading is implemented for the reader. If True, both
+        cases and memory mapping to file are tested.
+    **kwargs :
+        Arguments for the reader.
+
+    Returns
+    -------
+    raw : Instance of Raw
+        A preloaded Raw object.
+    """
+    tempdir = _TempDir()
+    raws = list()
+    raws.append(reader(**kwargs))
+    rng = np.random.RandomState(0)
+    picks = rng.permutation(np.arange(len(raws[0].ch_names)))[:10]
+    if test_preloading:
+        buffer_fname = op.join(tempdir, 'buffer')
+        raws.append(reader(preload=buffer_fname, **kwargs))
+        raws.append(reader(preload=True, **kwargs))
+
+        if test_blocks:
+            bnd = int(round(raws[0].info['buffer_size_sec'] *
+                            raws[0].info['sfreq']))
+        else:
+            bnd = raws[0].n_times // 2
+        slices = (slice(0, bnd), slice(bnd - 1, bnd), slice(3, bnd),
+                  slice(3, 300), slice(None))
+        if raws[0].n_times >= 2 * bnd:  # at least two complete blocks
+            slices = slices + (slice(bnd, 2 * bnd), slice(bnd, bnd + 1),
+                               slice(0, bnd + 100))
+        for sl_time in slices:
+            for raw in raws[1:]:
+                data1, times1 = raws[0][picks, sl_time]
+                data2, times2 = raw[picks, sl_time]
+                assert_allclose(data1, data2)
+                assert_allclose(times1, times2)
+
+    raw = raws[-1]  # use preloaded raw
+    full_data = raw._data
+
+    print(raw)  # to test repr
+    print(raw.info)  # to test Info reprs
+
+    # Test saving and reading
+    out_fname = op.join(tempdir, 'test_raw.fif')
+    for obj in raws:
+        obj.save(out_fname, overwrite=True, buffer_size_sec=1)
+        Raw(out_fname)
+        obj.save(out_fname, tmax=obj.times[-1], overwrite=True)
+        raw3 = Raw(out_fname)
+        assert_equal(sorted(raw.info.keys()), sorted(raw3.info.keys()))
+        assert_array_almost_equal(raw3.load_data()._data[0:20],
+                                  full_data[0:20])
+        assert_array_almost_equal(raw.times, raw3.times)
+
+        assert_true(not math.isnan(raw3.info['highpass']))
+        assert_true(not math.isnan(raw3.info['lowpass']))
```

Comment:
I added this test as before. EGI was setting highpass and lowpass to None. Then it was written to disk as float hence the Nan when reading back.